src/hotspot/share/opto/phaseX.cpp

```diff
@@ -1843,8 +1850,9 @@ void PhaseCCP::push_phis(Unique_Node_List& worklist, const Node* use) const {
 
 // If we changed the receiver type to a call, we need to revisit the Catch node following the call. It's looking for a
 // non-NULL receiver to know when to enable the regular fall-through path in addition to the NullPtrException path.
+// Same if true if the type of a ValidLengthTest input to an AllocateArrayNode changes
 void PhaseCCP::push_catch(Unique_Node_List& worklist, const Node* use) {
-  if (use->is_Call()) {
+  if (use->is_Call() || use->is_AllocateArray()) {
```

Comment:
hi, Roland, 
I understand your intention here, but isn't AllocateArrayNode also a CallNode? 
My understanding is that use->is_Call() is true if use is an AllocationNode.